projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1073,30 +1073,23 @@ function get_paywall_blocks( $newsletter_access_level ) {
 		? __( 'Already a paid subscriber?', 'jetpack' )
 		: __( 'Already a subscriber?', 'jetpack' );
 
-	$sign_in = '';
-	$host    = new Host();
-	if ( ! is_user_logged_in() && $host->is_wpcom_simple() ) {
-		$id           = '';
-		$sign_in_link = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
-		$button_text  = esc_html__( 'Log in', 'jetpack' );
-	} elseif ( class_exists( 'Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Jetpack_Token_Subscription_Service' ) &&
-		! $host->is_wpcom_simple() ) {
-		// We are on Jetpack and no cookie is set
-		$id              = 'jp_retrieve_subscriptions_link';
-		$sign_in_link    = '#'; // listening to "click" event in view.js
-		$access_question = __( 'I am already subscribed.', 'jetpack' );
-		if ( ! Jetpack_Token_Subscription_Service::has_token_from_cookie() ) {
-			$button_text = esc_html__( 'Log in', 'jetpack' );
+	$button_text     = esc_html__( 'Log in', 'jetpack' );
+	$sign_in_link    = 'https://subscribe.wordpress.com/memberships/jwt?site_id=' . \Jetpack_Options::get_option( 'id' ) . '&redirect_url=' . get_current_url();
+	$access_question = __( 'I am already subscribed.', 'jetpack' );
+
+	if ( is_user_auth() ) {
+		$button_text = esc_html__( 'Switch accounts', 'jetpack' );
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			// custom domain
+			$sign_in_link = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
 		} else {
```

Comment:
this "else" is not needed BTW: $sign_in_link is not modified from line 1077

Reply:
👍 good catch, forgot to delete it when I moved it up